pkg/infrastructure/aws/vpc.go

```diff
@@ -0,0 +1,1362 @@
+	if err != nil {
+		return nil, fmt.Errorf("cannot list internet gateways: %w", err)
+	}
+	for _, igw := range result.InternetGateways {
+		return igw, nil
+	}
+	return nil, nil
+}
+
+func (o *CreateInfraOptions) CreateWorkerSecurityGroup(l *logrus.Logger, client ec2iface.EC2API, vpcID string) (string, error) {
+	backoff := wait.Backoff{
+		Steps:    10,
+		Duration: 3 * time.Second,
+		Factor:   1.0,
+		Jitter:   0.1,
+	}
+	groupName := fmt.Sprintf("%s-worker-sg", o.InfraID)
+	securityGroup, err := o.existingSecurityGroup(client, groupName)
+	if err != nil {
+		return "", err
+	}
+	if securityGroup == nil {
+		result, err := client.CreateSecurityGroup(&ec2.CreateSecurityGroupInput{
+			GroupName:         aws.String(groupName),
+			Description:       aws.String("worker security group"),
+			VpcId:             aws.String(vpcID),
+			TagSpecifications: o.ec2TagSpecifications("security-group", groupName),
+		})
+		if err != nil {
+			return "", fmt.Errorf("cannot create worker security group: %w", err)
+		}
+		var sgResult *ec2.DescribeSecurityGroupsOutput
+		err = retry.OnError(backoff, func(error) bool { return true }, func() error {
+			var err error
+			sgResult, err = client.DescribeSecurityGroups(&ec2.DescribeSecurityGroupsInput{
+				GroupIds: []*string{result.GroupId},
+			})
+			if err != nil || len(sgResult.SecurityGroups) == 0 {
+				return fmt.Errorf("not found yet")
+			}
+			return nil
+		})
+		if err != nil {
+			return "", fmt.Errorf("cannot find security group that was just created (%s)", aws.StringValue(result.GroupId))
+		}
+		securityGroup = sgResult.SecurityGroups[0]
+		l.Info("Created security group", "name", groupName, "id", aws.StringValue(securityGroup.GroupId))
+	} else {
+		l.Info("Found existing security group", "name", groupName, "id", aws.StringValue(securityGroup.GroupId))
+	}
+	securityGroupID := aws.StringValue(securityGroup.GroupId)
+	//sgUserID := aws.StringValue(securityGroup.OwnerId)
+	egressPermissions := DefaultWorkerSGEgressRules()
+	ingressPermissions := DefaultWorkerSGIngressRules()
+
+	var egressToAuthorize []*ec2.IpPermission
+	var ingressToAuthorize []*ec2.IpPermission
+
+	for _, permission := range egressPermissions {
+		if !includesPermission(securityGroup.IpPermissionsEgress, permission) {
+			egressToAuthorize = append(egressToAuthorize, permission)
+		}
+	}
+
+	for _, permission := range ingressPermissions {
+		if !includesPermission(securityGroup.IpPermissions, permission) {
+			ingressToAuthorize = append(ingressToAuthorize, permission)
+		}
+	}
+
+	const duplicatePermissionErrorCode = "InvalidPermission.Duplicate"
+	if len(egressToAuthorize) > 0 {
+		err = retry.OnError(backoff, func(error) bool { return true }, func() error {
+			_, err := client.AuthorizeSecurityGroupEgress(&ec2.AuthorizeSecurityGroupEgressInput{
+				GroupId:       aws.String(securityGroupID),
+				IpPermissions: egressToAuthorize,
+			})
+			var awsErr awserr.Error
```

Comment:
I'm pretty sure this needs to be a pointer:
```suggestion
			var awsErr *awserr.Error
```